index.bs

```diff
@@ -563,75 +568,125 @@ This method takes the following parameters:
+    Otherwise:
     1. Let |effectiveDomain| be the |callerOrigin|'s [=effective domain=].
-    1. If |effectiveDomain| is null, then reject |promise| with a {{DOMException}} whose name is "{{SecurityError}}"
-        and terminate this algorithm.
-    1. If {{ScopedCredentialOptions/rpId}} [=is not a registrable domain suffix of and is not equal to=]
-        |effectiveDomain|, reject |promise| with a {{DOMException}} whose name is "{{SecurityError}}", and
+    1. If |effectiveDomain| is null, then return [=a promise rejected with=] a {{DOMException}} whose name is 
+        "{{SecurityError}}" and terminate this algorithm.
+    1. If {{AssertionOptions/rpId}} [=is not a registrable domain suffix of and is not equal to=]
+        |effectiveDomain|, return [=a promise rejected with=] a {{DOMException}} whose name is "{{SecurityError}}", and
         terminate this algorithm.
-    1. Set |rpId| to the {{ScopedCredentialOptions/rpId}}.
+    1. Set |rpId| to the {{AssertionOptions/rpId}}.
 
-1. If the {{AssertionOptions/extensions}} member of {{options}} is [=present=], process any extensions supported by this
-    client platform, to produce the extension data that needs to be sent to the authenticator. If an error is encountered while
-    processing an extension, skip that extension and do not produce any extension data for it. Call the result of this
-    processing |clientExtensions|.
+1. Let |clientExtensions| be a new [=list=].
 
-1. Use {{assertionChallenge}}, |callerOrigin| and |rpId|, along with the token binding key associated with |callerOrigin| (if
-    any), to create a {{ClientData}} structure representing this request. Choose a hash algorithm for {{ClientData/hashAlg}} and
-    compute the [=JSON-serialized client data=] |clientDataJSON| and the [=hash of the serialized client data=]
-    |clientDataHash|.
+1. If the {{AssertionOptions/extensions}} member of {{options}} is [=present=], then [=map/for each=]
+    |extension| → |argument| of <code>{{options}}.{{AssertionOptions/extensions}}</code>:
+    1. If |extension| is not supported by this client platform, then [=continue=].
+    1. Otherwise, let |result| be the result of running |extension|'s [=client processing=] algorithm on |argument|. If the
+        algorithm returned an error, [=continue=].
+    1. [=list/Append=] |result| to |clientExtensions|.
+
+1. Let |clientData| be a new {{ClientData}} instance whose fields are:
+    : {{challenge}}
+    :: The [=base64url encoding=] of {{assertionChallenge}}
+    : {{origin}}
+    :: The [=unicode serialization of an origin|unicode serialization=] of |rpId|
+    : {{hashAlg}}
+    :: The [=recognized algorithm name=] of the hash algorithm selected by the client for generating the
+        [=hash of the serialized client data=]
+    : {{tokenBinding}}
+    :: The [=Token Binding ID=] associated with |callerOrigin| (if any)
```

Comment:
`tokenBinding` is an optional member of the ClientData dictionary, so the implication here is it would not appear in the dictionary if a Token Binding ID is unanavailable, thus the `(if any)` qualification.  Perhaps we can make that more clear.